tests/integration_tests/functional/test_feat_parity.py

```diff
@@ -0,0 +1,407 @@
+        ), f"{cpu_template}: {kernel_ver=} {leaf=:#x} {subleaf=:#x} {reg=} {actual=:#x}, {expected=:#x}"
+
+    for leaf, subleaf, reg, flags in must_be_unset:
+        assert reg in allowed_regs
+        actual = cpuid[(leaf, subleaf, reg)] & flags
+        expected = 0
+        assert (
+            actual == expected
+        ), f"{cpu_template} {kernel_ver=} {leaf=:#x} {subleaf=:#x} {reg=} {actual=:#x}, {expected=:#x}"
+
+
+def _test_cpuid_feat_flags_matrix(
+    bin_cloner_path,
+    network_config,
+    cpu_template,
+    flags_must_be_set,
+    flags_must_be_unset,
+):
+    """
+    This launches tests matrix for CPUID feature flag checks for the given CPU template.
+    """
+
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Testing matrix:
+    # - Guest kernel: Linux 4.14 & Linux 5.10
+    # - Rootfs: Ubuntu 18.04 with msr-tools package installed
+    # - Microvm: 1vCPU with 1024 MB RAM
+    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="1vcpu_1024mb"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels())
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="bionic-msrtools"))
+    assert len(disk_artifacts) == 1
+
+    test_context = TestContext()
+    test_context.custom = {
+        "builder": MicrovmBuilder(bin_cloner_path),
+        "network_config": network_config,
+        "cpu_template": cpu_template,
+        "flags_must_be_set": flags_must_be_set,
+        "flags_must_be_unset": flags_must_be_unset,
+    }
+    test_matrix = TestMatrix(
+        context=test_context,
+        artifact_sets=[microvm_artifacts, kernel_artifacts, disk_artifacts],
+    )
+    test_matrix.run_test(_test_cpuid_feat_flags)
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64", reason="CPU templates are only available on x86_64."
+)
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor()
+    not in [cpuid_utils.CpuVendor.INTEL, cpuid_utils.CpuVendor.AMD],
+    reason="CPU templates are only available on Intel and AMD.",
+)
+@pytest.mark.parametrize(
+    "cpu_template",
+    ["C3", "T2", "T2S", "T2CL"]
+    if cpuid_utils.get_cpu_vendor() == cpuid_utils.CpuVendor.INTEL
+    else ["T2A"],
+)
+def test_feat_parity_cpuid_mpx(bin_cloner_path, network_config, cpu_template):
+    """
+    Verifies that MPX (Memory Protection Extensions) is not enabled in any of the supported CPU templates.
+
+    @type: functional
+    """
+    # fmt: off
+    must_be_set = []
+    must_be_unset = [
+        (0x7, 0x0, "ebx",
+            (1 << 14) # MPX
+        ),
+    ]
+    # fmt: on
+
+    _test_cpuid_feat_flags_matrix(
+        bin_cloner_path, network_config, cpu_template, must_be_set, must_be_unset
+    )
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64", reason="CPU templates are only available on x86_64."
+)
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor()
+    not in [cpuid_utils.CpuVendor.INTEL, cpuid_utils.CpuVendor.AMD],
+    reason="CPU templates are only available on Intel and AMD.",
+)
```

Comment:
Can we use a constant to simplify these?
```suggestion
CPUID_MANUFACTURER_SUPPORTED = cpuid_utils.get_cpu_vendor() in [cpuid_utils.CpuVendor.INTEL, cpuid_utils.CpuVendor.AMD]
@pytest.mark.skipif(
    not CPUID_MANUFACTURER_SUPPORTED ,
    reason="CPU templates are only available on Intel and AMD.",
)
```

Reply:
done